### PR TITLE
Doc updates for MSTest

### LINKED ISSUE #4157: Missing docs
- `--config-file` see #4124
- env vars to change locale of the platform #4122
- enable fakes from mstest sdk #4121
- RSP file #4072 
- Review analyzers and code fixes support

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes https://github.com/microsoft/testfx/issues/4157

@Evangelink Can you take a look please?

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/testing/mstest-analyzers/mstest0021.md](https://github.com/dotnet/docs/blob/24e68f5331775e76b35a49e46193d045782266dc/docs/core/testing/mstest-analyzers/mstest0021.md) | [MSTEST0021: Prefer Dispose over TestCleanup methods](https://review.learn.microsoft.com/en-us/dotnet/core/testing/mstest-analyzers/mstest0021?branch=pr-en-us-43786) |
| [docs/core/testing/unit-testing-platform-config.md](https://github.com/dotnet/docs/blob/24e68f5331775e76b35a49e46193d045782266dc/docs/core/testing/unit-testing-platform-config.md) | [Microsoft.Testing.Platform config options](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-config?branch=pr-en-us-43786) |
| [docs/core/testing/unit-testing-platform-extensions-fakes.md](https://github.com/dotnet/docs/blob/24e68f5331775e76b35a49e46193d045782266dc/docs/core/testing/unit-testing-platform-extensions-fakes.md) | [Fakes extension](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-extensions-fakes?branch=pr-en-us-43786) |
| [docs/core/testing/unit-testing-platform-intro.md](https://github.com/dotnet/docs/blob/24e68f5331775e76b35a49e46193d045782266dc/docs/core/testing/unit-testing-platform-intro.md) | [docs/core/testing/unit-testing-platform-intro](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-intro?branch=pr-en-us-43786) |

### CODE REVIEW

File: docs/core/testing/unit-testing-platform-config.md [hunk 1]
```diff
@@ -38,3 +40,7 @@ Environment variables can be used to supply some runtime configuration informati
 
 > [!NOTE]
 > Environment variables take precedence over configuration settings in the *testconfig.json* file.
+
+### `TESTINGPLATFORM_UI_LANGUAGE` environment variable
```
Comment (Youssef1313) [inline docs/core/testing/unit-testing-platform-config.md:44]: Is this the right doc page for documenting this? @Evangelink

File: docs/core/testing/unit-testing-platform-intro.md [hunk 2]
```diff
@@ -183,56 +183,72 @@ To run a test project in CI add one step for each test executable that you wish
...
 
-The prefix for the log file name. Defaults to `"log_"`.
+  The prefix for the log file name. Defaults to `"log_"`.
 
 - **`--diagnostic-verbosity`**
 
-Defines the verbosity level when the `--diagnostic` switch is used. The available values are `Trace`, `Debug`, `Information`, `Warning`, `Error`, or `Critical`.
+  Defines the verbosity level when the `--diagnostic` switch is used. The available values are `Trace`, `Debug`, `Information`, `Warning`, `Error`, or `Critical`.
+
+- **`--exit-on-process-exit`**
```
Comment (Youssef1313) [inline docs/core/testing/unit-testing-platform-intro.md:214]: @Evangelink I wanted to make sure if we want to document this one, or if it was intentionally missing.

File: docs/core/testing/unit-testing-platform-intro.md [hunk 3]
```diff
@@ -183,56 +183,72 @@ To run a test project in CI add one step for each test executable that you wish
...
 
-Specifies the minimum number of tests that are expected to run. By default, at least one test is expected to run.
+  Specifies the minimum number of tests that are expected to run. By default, at least one test is expected to run.
 
 - **`--results-directory`**
 
-The directory where the test results are going to be placed. If the specified directory doesn't exist, it's created. The default is `TestResults` in the directory that contains the test application.
+  The directory where the test results are going to be placed. If the specified directory doesn't exist, it's created. The default is `TestResults` in the directory that contains the test application.
+
+- **`--timeout`**
```
Comment (Youssef1313) [inline docs/core/testing/unit-testing-platform-intro.md:249]: @Evangelink Same here. Do we want to document this? Or was it missing intentionally?

File: docs/core/testing/unit-testing-platform-intro.md (hunk 2 above)
Comment (Evangelink) [reply docs/core/testing/unit-testing-platform-intro.md:214]: Good question, at the moment we mainly use it to ensure that if some other launching process terminates we also terminates. It feels more technical but maybe it's worth mentioning for users.

WDYT @MarcoRossignoli ?

File: docs/core/testing/unit-testing-platform-intro.md (hunk 3 above)
Comment (Evangelink) [reply docs/core/testing/unit-testing-platform-intro.md:249]: Definitely a miss!

File: docs/core/testing/unit-testing-platform-config.md (hunk 1 above)
Comment (Youssef1313) [reply docs/core/testing/unit-testing-platform-config.md:44]: @Evangelink This is the docs for `DOTNET_CLI_UI_LANGUAGE`
https://github.com/dotnet/docs/blob/3208829cb687c603ab97762f8cfad9aa12dd3c4c/docs/core/tools/dotnet-environment-variables.md?plain=1#L335-L337

Comment (Evangelink) [reply docs/core/testing/unit-testing-platform-config.md:44]: We can copy the first part:

```
Sets the language of the CLI UI using a locale value such as `en-us`. The supported values are the same as for Visual Studio. For more information, see the section on changing the installer language in the [Visual Studio installation documentation](/visualstudio/install/install-visual-studio).
```

In the `dotnet test` integration doc we could link to this part of the CLI doc.

File: docs/core/testing/unit-testing-platform-intro.md (hunk 2 above)
Comment (MarcoRossignoli) [reply docs/core/testing/unit-testing-platform-intro.md:235]: I would document it, it's public right? And could be useful if the test is started from some tool.

File: docs/core/testing/unit-testing-platform-intro.md [hunk 4]
```diff
@@ -183,56 +183,93 @@ To run a test project in CI add one step for each test executable that you wish
 
 The list below described only the platform options. To see the specific options brought by each extension, either refer to the extension documentation page or use the `--help` option.
 
+- **`@`**
+
+  Specifies a response file (rsp). A response file is a file that includes command-line arguments, each in a separate line. This helps for very long commands that may exceed the terminal limits. You can also specify a response file along with extra command-line arguments. For example:
```
Comment (Evangelink) [inline docs/core/testing/unit-testing-platform-intro.md:188]: Please add either that text directly or a reword of it:

```
Specifies the name of the response file. The response file name must immediately follow the @ character with no white space between the @ character and the response file name.

Options in a response file are interpreted as if they were present at that place in the command line. Each argument in a response file must begin and end on the same line. You cannot use the backslash character (\) to concatenate lines.
```

**Review by Evangelink — APPROVED** [review]
A small request but LGTM

File: docs/core/testing/mstest-analyzers/mstest0021.md [hunk 5]
```diff
@@ -22,7 +22,7 @@ ms.author: amauryleve
 | **Enabled by default**              | No                                      |
 | **Default severity**                | Info                                    |
 | **Introduced in version**           | 3.4.0                                   |
-| **There is a code fix**             | No                                      |
+| **There is a code fix**             | Yes, starting with 3.7.0                |
```
Comment (gewarren) [inline docs/core/testing/mstest-analyzers/mstest0021.md:25]:
```suggestion
| **Is there a code fix**             | Yes, starting with 3.7.0                |
```

File: docs/core/testing/unit-testing-platform-config.md [hunk 6]
```diff
@@ -29,6 +29,8 @@ The platform will automatically detect and load the *[appname].testconfig.json*
 
 When using [Microsoft.Testing.Platform.MSBuild](https://www.nuget.org/packages/Microsoft.Testing.Platform.MSBuild), you can simply create a *testconfig.json* file that will be automatically renamed to *[appname].testconfig.json* and moved to the output directory of the test project.
 
+Starting with Microsoft.Testing.Platform 1.5, the command-line argument `--config-file` can be used to specify the path to the *testconfig.json*. This takes precedence over the *[appname].testconfig.json* file.
```
Comment (gewarren) [inline docs/core/testing/unit-testing-platform-config.md:32]:
```suggestion
Starting with Microsoft.Testing.Platform 1.5, you can use the command-line argument `--config-file` to specify the path to the *testconfig.json*. This file takes precedence over the *[appname].testconfig.json* file.
```

File: docs/core/testing/unit-testing-platform-config.md [hunk 7]
```diff
@@ -38,3 +40,7 @@ Environment variables can be used to supply some runtime configuration informati
 
 > [!NOTE]
 > Environment variables take precedence over configuration settings in the *testconfig.json* file.
+
+### `TESTINGPLATFORM_UI_LANGUAGE` environment variable
+
+Starting with Microsoft.Testing.Platform 1.5, this environment variable sets the language of the platform for displaying messages and logs using a locale value such as `en-us`. This takes precedence over the Visual Studio and .NET SDK languages. The supported values are the same as for Visual Studio. For more information, see the section on changing the installer language in the [Visual Studio installation documentation](/visualstudio/install/install-visual-studio).
```
Comment (gewarren) [inline docs/core/testing/unit-testing-platform-config.md:46]:
```suggestion
Starting with Microsoft.Testing.Platform 1.5, this environment variable sets the language of the platform for displaying messages and logs using a locale value such as `en-us`. This language takes precedence over the Visual Studio and .NET SDK languages. The supported values are the same as for Visual Studio. For more information, see the section on changing the installer language in the [Visual Studio installation documentation](/visualstudio/install/install-visual-studio).
```

File: docs/core/testing/unit-testing-platform-extensions-fakes.md [hunk 8]
```diff
@@ -25,3 +25,5 @@ To use the new extension with an existing project, update the existing `Microsof
 - </Reference>
 + <PackageReference Include="Microsoft.Testing.Extensions.Fakes" Version="17.11.0-beta.24319.3" />
 ```
+
+If you are using MSTest.Sdk 3.7 or later, use `EnableMicrosoftTestingExtensionsFakes` property to enable the extension and don't add the package reference.
```
Comment (gewarren) [inline docs/core/testing/unit-testing-platform-extensions-fakes.md:29]:
```suggestion
If you are using MSTest.Sdk 3.7 or later, use the `EnableMicrosoftTestingExtensionsFakes` property to enable the extension and don't add the package reference.
```

File: docs/core/testing/unit-testing-platform-intro.md (hunk 4 above)
Comment (gewarren) [inline docs/core/testing/unit-testing-platform-intro.md:188]:
```suggestion
  Specifies a response file (rsp). A response file is a file that includes command-line arguments, each in a separate line. This helps for very long commands that might exceed the terminal limits. You can also specify a response file along with extra command-line arguments. For example:
```

File: docs/core/testing/unit-testing-platform-intro.md [hunk 9]
```diff
@@ -183,56 +183,93 @@ To run a test project in CI add one step for each test executable that you wish
...
-Displays advanced information about the .NET Test Application such as:
+  Displays advanced information about the .NET Test Application such as:
 
-- The platform.
-- The environment.
-- Each registered command line provider, such as its, `name`, `version`, `description` and `options`.
-- Each registered tool, such as its, `command`, `name`, `version`, `description`, and all command line providers.
+  - The platform.
+  - The environment.
+  - Each registered command line provider, such as its, `name`, `version`, `description` and `options`.
```
Comment (gewarren) [inline docs/core/testing/unit-testing-platform-intro.md:253]:
```suggestion
  - Each registered command line provider, such as its `name`, `version`, `description`, and `options`.
```

File: docs/core/testing/unit-testing-platform-intro.md [hunk 10]
```diff
@@ -183,56 +183,93 @@ To run a test project in CI add one step for each test executable that you wish
...
+  - Each registered tool, such as its, `command`, `name`, `version`, `description`, and all command line providers.
```
Comment (gewarren) [inline docs/core/testing/unit-testing-platform-intro.md:254]:
```suggestion
  - Each registered tool, such as its `command`, `name`, `version`, `description`, and all command-line providers.
```

**Review by gewarren — APPROVED** [review]

**Review by Evangelink — APPROVED** [review]

**Review by Evangelink — APPROVED** [review]

**Pull request merged**